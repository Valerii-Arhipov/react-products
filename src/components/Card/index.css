.card {
    display: flex;
    border: 1px solid black;
    margin-bottom: 20px;
}
.card__photo {
    width: 200px;
    height: auto;
}
.card__info-block {
    padding: 10px;
}
.card__arrow {
    width: 20px;
    height: auto;
}
.card__votes-block {
    display: flex;
    align-items: center;
    min-width: 70px;
}
.card__text {
    font-size: 14px;
    text-align: left;
}
.card__text--count {
    font-weight: bold;
    color: cornflowerblue;
}
.card__text--title {
    font-weight: bold;
}
.card__text--submitted {
    color: darkgrey;
    margin-right: 10px;
}
.card__avatar {
    height: 30px;
    width: 30px;
    border-radius: 50%;
}
