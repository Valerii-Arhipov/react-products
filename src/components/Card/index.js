import React, { memo } from 'react';
import PropTypes from 'prop-types';

import './index.css';

function areEqual(prevProps, nextProps) {
  return prevProps.product.votes === nextProps.product.votes;
}

function Card({
  changeVote, product: {
    id,
    title,
    description,
    votes,
    submitterAvatarUrl,
    productImageUrl,
  },
}) {
  return (
    <div className="card">
      <img src={productImageUrl} className="card__photo" alt="product" />
      <div className="card__info-block">
        <div className="card__votes-block">
          <img
            aria-hidden="true"
            onClick={() => changeVote(id)}
            className="card_arrow"
            src="images/ui/arrow.png"
            alt="arrow"
          />
          <p className="card__text card__text--count">{votes}</p>
        </div>
        <p className="card__text card__text--title">{title}</p>
        <p className="card__text">{description}</p>
        <div className="card__votes-block">
          <p className="card__text card__text--submitted">Submitted by:</p>
          <img src={submitterAvatarUrl} className="card__avatar" alt="avatar" />
        </div>

      </div>
    </div>
  );
}

Card.propTypes = {
  changeVote: PropTypes.func.isRequired,
  product: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    description: PropTypes.string,
    url: PropTypes.string,
    votes: PropTypes.number,
    submitterAvatarUrl: PropTypes.string,
    productImageUrl: PropTypes.string,

  }).isRequired,
};

export default memo(Card, areEqual);
