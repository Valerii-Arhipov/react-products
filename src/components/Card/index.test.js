import React from 'react';
import renderer from 'react-test-renderer';
import Card from './index';

const testProduct = {
  id: 1,
  title: 'Yellow Pail',
  description: 'On-demand sand castle construction expertise.',
  url: '#',
  votes: 40,
  submitterAvatarUrl: 'images/avatars/daniel.jpg',
  productImageUrl: 'images/products/image-aqua.png',
};

test('renders correctly', () => {
  const tree = renderer
    .create(<Card product={testProduct} changeVote={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
