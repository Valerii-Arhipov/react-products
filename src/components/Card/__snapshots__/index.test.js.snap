// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card"
>
  <img
    alt="product"
    className="card__photo"
    src="images/products/image-aqua.png"
  />
  <div
    className="card__info-block"
  >
    <div
      className="card__votes-block"
    >
      <img
        alt="arrow"
        aria-hidden="true"
        className="card_arrow"
        onClick={[Function]}
        src="images/ui/arrow.png"
      />
      <p
        className="card__text card__text--count"
      >
        40
      </p>
    </div>
    <p
      className="card__text card__text--title"
    >
      Yellow Pail
    </p>
    <p
      className="card__text"
    >
      On-demand sand castle construction expertise.
    </p>
    <div
      className="card__votes-block"
    >
      <p
        className="card__text card__text--submitted"
      >
        Submitted by:
      </p>
      <img
        alt="avatar"
        className="card__avatar"
        src="images/avatars/daniel.jpg"
      />
    </div>
  </div>
</div>
`;
